### PR TITLE
Add note about Faces being a private API

### PR DESCRIPTION
**Pull request by nfagerlund**

### CODE REVIEW

File: source/puppet/4.3/reference/deprecated_api.markdown [hunk 1]
```diff
@@ -6,19 +6,27 @@ title: "Deprecated Extension Points and APIs"
 
 The following APIs and extension points are deprecated, and will be removed in Puppet 5.0.
 
+## Puppet Faces is a private API
 
-Resource Type and Provider APIs
------
+(This isn't actually a deprecation, since we don't have a specific plan to remove the Faces APIs. But we needed a place to mention this on the record, and the deprecations section seemed like the best long-term bet.)
 
+The `Puppet::Face` and `Puppet::Indirector::Face` classes are private APIs, and not particularly good ones. Please don't use them. If you want to develop CLI applications that use Puppet's data, you should:
```
Comment (ahpook) [inline source/puppet/4.3/reference/deprecated_api.markdown]: s/not particularly good ones/not well-documented or supported ones/

File: source/puppet/4.3/reference/deprecated_api.markdown [hunk 2]
```diff
@@ -6,19 +6,27 @@ title: "Deprecated Extension Points and APIs"
...
+
+* Pick a popular, well-maintained Ruby framework for CLI apps, like Thor or Commander.
+* Use Puppet's public Ruby APIs to access its settings and data. See the [developer documentation](./yard/frames.html) for comprehensive info about the classes and methods available for this.
+* Distribute your application as a Gem or native package, not as a Puppet module.
+* If you _really_ want your command to be available as a `puppet <NAME>`-style subcommand, you can name it `puppet-<NAME>` --- there's a secret feature that lets you leave out the hyphen.
```
Comment (ahpook) [inline source/puppet/4.3/reference/deprecated_api.markdown]: "and have your gem or package install it in $PATH"?  (or does it have to be in a special directory?)

File: source/puppet/4.3/reference/deprecated_api.markdown [hunk 3]
```diff
@@ -6,19 +6,27 @@ title: "Deprecated Extension Points and APIs"
...
+
+At one point, we had vague ambitions to turn `Puppet::Face` into a robust general-purpose CLI framework, but that never ended up happening. We aren't putting any effort into maintaining or improving it, and our upcoming tools won't be built on it.
```
Comment (ahpook) [inline source/puppet/4.3/reference/deprecated_api.markdown]: X this graf

**Comment by ahpook** [comment]
Couple of comments on the first of the two copies, but otherwise :+1:  Thanks Nick

**Comment by MFredette** [comment]
Do we need this paragraph: At one point, we had vague ambitions to turn `Puppet::Face` into a robust general-purpose CLI framework, but that never ended up happening. We aren't putting any effort into maintaining or improving it, and our upcoming tools won't be built on it.

Seems like we could lose it.

**Comment by nfagerlund** [comment]
Addressed all the comments, gonna merge and push. Thanks!

**Pull request merged**